src/hubAuth.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'path';
+import * as os from 'os';
+import { fs } from '@salesforce/core';
+import * as shell from 'shelljs';
+import { debug } from 'debug';
+
+// this seems to be a known eslint error for enums
+// eslint-disable-next-line no-shadow
+enum AuthStrategy {
+  JWT = 'JWT',
+  AUTH_URL = 'AUTH_URL',
+  REUSE = 'REUSE',
+  NONE = 'NONE',
+}
+
+export const authFromStubbedHome = (homeDir: string): void => {
+  const logger = debug('testkit:authFromStubbedHome');
+  if (getAuthStrategy() === AuthStrategy.JWT && process.env.TESTKIT_JWT_KEY) {
+    logger('trying jwt auth');
+    const jwtKey = path.join(homeDir, 'jwtKey');
+    fs.writeFileSync(jwtKey, process.env.TESTKIT_JWT_KEY);
+
+    shell.exec(
+      `sfdx auth:jwt:grant -d -u ${process.env.TESTKIT_HUB_USERNAME} -i ${
+        process.env.TESTKIT_JWT_CLIENT_ID
+      } -f ${jwtKey} -r ${process.env.TESTKIT_HUB_INSTANCE || 'https://login.salesforce.com'}`,
+      { silent: true }
+    );
+    return;
+  }
+  if (getAuthStrategy() === AuthStrategy.AUTH_URL && process.env.TESTKIT_AUTH_URL) {
+    logger('trying to authenticate with AuthUrl');
+
+    const tmpUrl = path.join(homeDir, 'tmpUrl');
+    fs.writeFileSync(tmpUrl, process.env.TESTKIT_AUTH_URL);
+
+    const shellOutput = shell.exec(`sfdx auth:sfdxurl:store -d -f ${tmpUrl}`, { silent: true });
+    logger(shellOutput);
```

Comment:
@mshanemc 
Since we are depending on `@salesforce/core` anyway why don't we create an AuthInfo directly and write it rather than write the auth url file and exec the command?  That seems a little cleaner and inline with the purpose of the core lib.  The auth plugin will have NUTs that test the auth commands.

Reply:
I saw free coverage of commands (and the convenience of not rewriting a subset of the auth commands here, too) and took it.  It should be the same either way.

Do you want the same no-shell approach for jwt?

Reply:
[for posterity: dropped because core requires async]